commons/src/main/java/io/hyscale/commons/models/DockerCredHelper.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.commons.models;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.hyscale.commons.commands.CommandExecutor;
+import io.hyscale.commons.constants.ToolConstants;
+import io.hyscale.commons.utils.ObjectMapperFactory;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Provides credential from the docker-credential-helper.Takes inputs @helperFuntion ie..credential helper name and registry name generates
+ * the "docker-credential-helperFuntion get" command and gets credentials if exists.
+ */
+public class DockerCredHelper {
+
+    private static final Logger logger = LoggerFactory.getLogger(CommandExecutor.class);
+    private static final String ECHO = "echo";
+    private static final String DOCKER_CREDENTIAL = "docker-credential-";
+    private static final String PIPE = " | ";
+    private static final String GET = "get";
+
+    private String helperFunction;
+
+    public String getHelperFunction() {
+        return helperFunction;
+    }
+
+    public DockerCredHelper(String helperFunction) {
+        this.helperFunction = helperFunction;
+    }
+
+    /**
+     * Fetches the credentials from credstore given registryUrl
```

Comment:
Add a reference about how you are fetching the credentials 
like
docker-credential-<> get 
stdinput of registry detailing